src/python/pants/init/parsed_specs_calculator.py

```diff
@@ -21,26 +27,34 @@
 logger = logging.getLogger(__name__)
 
 
-class TargetRootsCalculator:
-  """Determines the target roots for a given pants run."""
+class InvalidSpecConstraint(Exception):
+  """Raised when invalid constraints are given via specs and arguments like --changed*."""
+
+
+class ParsedSpecsCalculator:
+  """Determines the parsed specs for a given Pants run."""
```

Comment:
All specs are parsed, no? "Parsed" here (and elsewhere) feels extraneous.

Reply:
Hm, do you think we should call it simply `Specs`? I had two reasons for using `ParsedSpecs`:
1) Distinguish between raw spec strings vs. parsed specs
   * But, raw specs are much more limited in their usage. Probably better for `Specs` to default to meaning the parsed `AddressSpecs` and `FilesystemSpecs` and exclusively using `raw specs` or `spec strings` to describe the unparsed arguments provided by users.
2) Reserve the word `Spec` because we aren't yet sure if that should exist and what it should look like.
    * However, after thinking about it, we probably would end up having a `Spec` superclass for one single FilesystemSpec vs. AddressSpec, but implement `Specs` as a dataclass just like `ParsedSpecs` is now, rather than as a `Collection[Spec]`.

So, I think you're right to call it Specs.